dbms/src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -523,17 +525,51 @@ BlockIO InterpreterCreateQuery::createTable(ASTCreateQuery & create)
               * Otherwise, concurrent queries for creating a table, if the table does not exist,
               *  can throw an exception, even if IF NOT EXISTS is specified.
               */
-            guard = context.getDDLGuardIfTableDoesntExist(
-                database_name,
-                table_name,
-                "Table " + database_name + "." + table_name + " is creating or attaching right now");
+            try
+            {
+                guard = context.getDDLGuardIfTableDoesntExist(
+                    database_name,
+                    table_name,
+                    "Table " + database_name + "." + table_name + " is creating or attaching right now");
 
-            if (!guard)
+                if (!guard)
+                {
+                    if (create.if_not_exists)
+                        return {};
+                    else
+                        throw Exception("Table " + database_name + "." + table_name + " already exists.", ErrorCodes::TABLE_ALREADY_EXISTS);
+                }
+            }
+            catch (Exception & e)
             {
-                if (create.if_not_exists)
-                    return {};
+                // Help: please check the logical could be ok
+                // Due to even if it throws this two error code, it can't ensure the table is completely created
+                // So we have to wait for the table created completely, then return to use the table.
+                // Thus, we choose to do a retry here to wait the table created completed.
+                if (e.code() == ErrorCodes::TABLE_ALREADY_EXISTS || e.code() == ErrorCodes::DDL_GUARD_IS_ACTIVE)
+                {
+                    LOG_ERROR(Logger::get("InterpreterCreateQuery"), "InterpreterCreateQuery::createTable failed, with error code is {}, error info is {}, stack_info is {}", e.code(), e.displayText(), e.getStackTrace().toString());
+                    for (int i = 0; i < 20; i++)
+                    { // retry for 1 mins
+                        while (!context.isTableExist(database_name, table_name))
+                        {
+                            const int wait_useconds = 20000;
+                            LOG_ERROR(
+                                Logger::get("InterpreterCreateQuery"),
+                                "InterpreterCreateQuery::createTable failed but table not exist now, \nWe will sleep for {}"
+                                " ms and try again.",
+                                wait_useconds / 1000);
+                            usleep(wait_useconds); // sleep 20ms
+                        }
+                        return {};
+                    }
```

Comment:
This part looks weird. If `isTableExist` keeps `false`, then the `while` will never end.

Reply:
And the `for` seems useless. It cannot help the goal "retry for 1 mins"

Reply:
There is a mistake for while, actually it should be a `if`, and then the for statement can help to fulfill the goal "retry for 1 mins"